### ISSUE TITLE
Default text size is too small

### ISSUE THREAD

**Issue by noahmotion** [tidyverse/ggplot2#3139]
The default size of axis, tick, and legend labels is too small. I expect the default size of these labels to be legible when figures are embedded in papers and presentations, but they almost never are.

Any ggplot figure with the default axis and tick label size can serve as a reprex, so:

```r
df = data.frame(x=rnorm(100),y=rnorm(100),g=c(rep("A",50),rep("B",50)))
png("test.png",width=500,height=500)
ggplot(df, mapping=aes(x=x, y=y, color=g)) + geom_point()
dev.off()
```

In Rstudio, if I make a figure and then export it, it's slightly better, but it's still not particularly good.

I don't know ggplot well enough to know how to change the code and submit a pull request, but I imagine that for someone who knows it well, it would be trivially easy to do. The package does so much so well, but the axis, tick, and legend label size is a persistent problem.

**Comment by clauswilke** [comment]
The size specifications are entirely meaningless unless you also specify a resolution at which you want to output your figure. Increase the resolution until you have output you're happy with. The following seems fine to me.

``` r
df = data.frame(x=rnorm(100),y=rnorm(100),g=c(rep("A",50),rep("B",50)))
png("test.png",width=500,height=500, res = 120)
ggplot(df, mapping=aes(x=x, y=y, color=g)) + geom_point()
dev.off()
```
![test](https://user-images.githubusercontent.com/4210929/52757500-75564c80-2fca-11e9-840d-b6408bd6a80e.png)

You can also increase the font size within a theme, e.g. by adding `theme_grey(14)`, or you can set this globally, via `theme_set(theme_grey(14))`, but it doesn't change the fact that the font size is only meaningful in the context of a specified output dimension. In general, with the default ggplot2 settings, figures that are about 4 by 3 inches look good. (The example above is ~4 by 4 inches. It's a little too tall.)

**Comment by noahmotion** [comment]
Fair point, but what is the default resolution for functions like `png()`? Whatever it is, it doesn't work well with the default axis/tick text size, and in my experience, many people don't bother to specify either such that the axis, tick, or legend labels are legible. It seems to me that the default text size in ggplot should work well with the default figure size and resolution.

**Comment by thomasp85** [comment]
Please see the documentation for png for further information about the device. The fact that the defaults doesn’t exactly suit your personal predominant usecase is not an argument for changing them - especially when it is so easy to change the relative font size on the fly with the theming system as well as define your own personal defaults

**Comment by noahmotion** [comment]
I have found that, easy or not, people _don't_ typically change tick/axis/legend label text size, and the majority of ggplot figures I see in papers, presentations, etc... have illegibly small tick/axis/legend labels. Given that ggplot has pretty good defaults for more or less everything else, I figured it might be better to lobby for this default to be changed rather than lobbying individual researchers to remember to specify tick/axis/legend label text size. For what it's worth, I'm coming at this as a consumer of ggplot figures, not a producer, since I _do_ specify these sorts of parameters when I think it's necessary to do so.
